src/errors.test.ts

```diff
@@ -164,6 +164,41 @@ describe('wrapError', () => {
     });
   });
 
+  describe('if the original error has a length === 2', () => {
+    it('returns an error with cause', () => {
+      // Save original Error
+      const OriginalError = global.Error;
+
+      // Create a mock error with a cause
+      class MockError extends Error {
+        constructor(message: string, options: { cause?: Error } = {}) {
+          super(message);
+          this.cause = options.cause;
+        }
+        cause: Error | undefined;
+      }
+      // Set length to 2
+      Object.defineProperty(MockError, 'length', { value: 2 });
+
+      // Replace global Error with MockError
+      global.Error = MockError as unknown as ErrorConstructor;
```

Comment:
I'd recommend doing this using Jest instead so that it gets automatically cleaned up. That saves you from having to manually restore, but more importantly it ensures the cleanup always happens even if the test fails partway through

```suggestion
      jest.replaceProperty(global, 'Error', MockError);
```

Reply:
ooof this was added in jest 29.4 and we are on 29.2 here. Ill try bumping it?

Reply:
Ive added a comment to remind us to change this liine when we upgrade jest. Hope thats okay for now.

Reply:
Have you tried bumping `@types/jest` as well?

Reply:
Sure, we can come back to it